eda/src/autogluon/eda/analysis/anomaly.py

```diff
@@ -0,0 +1,342 @@
+                self._train_index_to_detector = {**self._train_index_to_detector, **{idx: i for idx in x_train.index}}
+                val_scores = detector.decision_function(x_val)  # outlier scores
+            folds_scores.append(pd.Series(name="score", data=val_scores, index=x_val.index))
+        return pd.concat(folds_scores, axis=0)[x.index]
+
+    def transform(self, x: pd.DataFrame):
+        """
+        Predict anomaly scores for the provided inputs.
+        This method uses average of scores produced by all the detectors trained on folds.
+
+        Parameters
+        ----------
+        x: pd.DataFrame
+            data to score; must be already pre-processed; numeric columns must have NaNs filled
+
+        Returns
+        -------
+        anomaly scores for the passed data
+        """
+        assert self._detectors is not None, "Detector is not fit - call `fit_transform` before calling `transform`"
+
+        folds_scores = []
+        for detector in self._detectors:
+            with _suod_silent_print(self.silent):
+                y_test_scores = detector.decision_function(x[self.original_features])
+            folds_scores.append(pd.DataFrame({"score": y_test_scores}, index=x.index))
+        score = pd.concat([df.score for df in folds_scores], axis=1).mean(axis=1)
+        score.name = "score"
+
+        return score[x.index]
+
+    def predict(self, x):
+        """
+        API-compatibility wrapper for :py:meth:`~autogluon.eda.analysis.anomaly.AnomalyDetector.transform`
+        """
+        return self.transform(x)
+
+
+class AnomalyDetectorAnalysis(AbstractAnalysis):
+    """
+    Anomaly detection analysis.
+
+    The analysis automatically creates cross-validation splits and fits detectors on each of them using
+    `train_data` input. The scores produced for the training data are produced using out-of-folds predictions.
+    All other datasets scores are produced using average of scores from detectors trained on individual folds (bag).
+
+    Please note, the analysis expects the data is ready to for fitting; all numeric columns must not have NaNs.
+    Pre-processing can be performed using :py:class:`~autogluon.eda.analysis.transform.ApplyFeatureGenerator`
+    and :py:class:`~autogluon.eda.analysis.dataset.ProblemTypeControl` (see example for more details).
+
+    State attributes
+
+    - `anomaly_detection.scores.<dataset>`
+        scores for each of the datasets passed into analysis (i.e. `train_data`, `test_data`)
+    - `anomaly_detection.explain_rows_fns.<dataset>`
+        if `store_explainability_data=True`, then analysis will store helper functions into this
+        variable. The function can be used later via :py:meth:`~autogluon.eda.auto.simple.explain_rows`
+        and automatically pre-populates `train_data`, `model` and `rows` parameters when called
+        (see example for more details)
+
+
+    Parameters
+    ----------
+    n_folds: int, default = 5
```

Comment:
Just curious, why 5 folds

Reply:
80% data for the training. Not too many folds to train.